Lib/ipaddress.py

```diff
@@ -1834,8 +1854,23 @@ def __init__(self, address):
         addr_str = str(address)
         if '/' in addr_str:
             raise AddressValueError("Unexpected '/' in %r" % address)
+        addr_str, self._scope_id = self._split_scope_id(addr_str)
+
         self._ip = self._ip_int_from_string(addr_str)
 
+    def __str__(self):
+        ip_str = super().__str__()
+        return ip_str if self.scope_id is None else ip_str + '%' + self.scope_id
```

Comment:
This is simply reversing the conditional so that None doesn't have to be explicitly specified: 
``a if x is None else b`` == ``b if x else a``

```suggestion
        return ip_str + '%' + self.scope_id if self.scope_id else ip_str
```

This is another shorter alternative, but the previous one might be a bit easier to read:
```python
return ip_str + ('%' + self.scope_id if self.scope_id else '')
```

Reply:
https://github.com/python/cpython/blob/033f0313013775ba841a8d965e32037469ea727d/Lib/ipaddress.py#L1799-L1803
Here I tried to exclude cases with blank `scope_id`, but your proposal is really reasonable if someone sets protected `self._scope_id` an empty value.